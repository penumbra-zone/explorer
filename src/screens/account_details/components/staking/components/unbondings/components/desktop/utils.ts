export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'validator',
    width: 25,
  },
  {
    key: 'commission',
    align: 'right',
    width: 25,
  },
  {
    key: 'amount',
    width: 25,
    align: 'right',
  },
  {
    key: 'linkedUntil',
    width: 25,
    align: 'right',
  },
];
