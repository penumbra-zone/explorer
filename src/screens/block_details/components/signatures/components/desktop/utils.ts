export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'validator',
    width: 34,
  },
  {
    key: 'votingPower',
    align: 'right',
    width: 33,
  },
  {
    key: 'signed',
    width: 33,
  },
];
