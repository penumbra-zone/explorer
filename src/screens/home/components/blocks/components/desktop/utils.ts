export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
}[] = [
  {
    key: 'height',
  },
  {
    key: 'validator',
  },
  {
    key: 'hash',
  },
  {
    key: 'txs',
  },
  {
    key: 'time',
    align: 'right',
  },
];
