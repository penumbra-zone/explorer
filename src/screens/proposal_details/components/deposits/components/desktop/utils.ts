export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'depositor',
    width: 34,
  },
  {
    key: 'amount',
    width: 33,
    align: 'right',
  },
  {
    key: 'time',
    width: 33,
  },
];
