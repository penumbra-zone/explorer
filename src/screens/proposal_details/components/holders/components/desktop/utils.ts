export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'rank',
    width: 10,
  },
  {
    key: 'address',
    width: 30,
  },
  {
    key: 'quantity',
    width: 20,
    align: 'right',
  },
  {
    key: 'percentage',
    width: 20,
  },
  {
    key: 'value',
    align: 'right',
    width: 20,
  },
];
