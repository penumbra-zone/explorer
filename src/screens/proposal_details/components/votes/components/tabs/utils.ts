export const tabLabels = [
  {
    key: 'all',
    num: 30,
  },
  {
    key: 'yes',
    num: 3,
  },
  {
    key: 'no',
    num: 3,
  },
  {
    key: 'noWithVeto',
    num: 3,
  },
];
