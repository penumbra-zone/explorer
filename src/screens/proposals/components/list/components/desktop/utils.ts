export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'idx',
    width: 15,
  },
  {
    key: 'token',
    width: 10,
  },
  {
    key: 'price',
    align: 'right',
    width: 15,
  },
  {
    key: 'change',
    align: 'right',
    width: 15,
  },
  {
    key: 'volume',
    align: 'right',
    width: 15,
  },
  {
    key: 'marketCap',
    align: 'right',
    width: 15,
  },
  {
    key: 'holders',
    align: 'right',
    width: 15,
  },
];
