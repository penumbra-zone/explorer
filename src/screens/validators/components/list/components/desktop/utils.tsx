import React from 'react';
import {
  Typography,
} from '@material-ui/core';
import { InfoPopover } from '@components';

export const fetchColumns = (t): {
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
  component?: React.ReactNode;
  sort?: boolean;
}[] => {
  return ([
    {
      key: 'idx',
      width: 10,
    },
    {
      key: 'validator',
      width: 20,
      sort: true,
    },
    {
      key: 'stake',
      align: 'right',
      width: 15,
      sort: true,
    },
    {
      key: 'fee',
      align: 'right',
      width: 10,
      sort: true,
    },
    {
      key: 'lastVote',
      align: 'right',
      width: 15,
    },
    {
      key: 'skipRate',
      width: 15,
      sort: true,
    },
    {
      key: 'condition',
      align: 'center',
      width: 15,
      component: (
        <Typography variant="h4" className="label popover">
          {t('condition')}
          <InfoPopover
            content={t('conditionExplanation')}
          />
        </Typography>
      ),
    },
  ]);
};
