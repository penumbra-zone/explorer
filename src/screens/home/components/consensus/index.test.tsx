import React from 'react';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import Consensus from '.';

// ==================================
// mocks
// ==================================
const mockI18n = {
  t: (key: string) => key,
  lang: 'en',
};
jest.mock('next-translate/useTranslation', () => () => mockI18n);

jest.mock('@components', () => ({
  Box: (props) => <div id="box" {...props} />,
  AvatarName: (props) => <div id="AvatarName" {...props} />,
}));

jest.mock('recharts', () => ({
  RadialBarChart: (props) => <div id="RadialBarChart" {...props} />,
  PolarAngleAxis: (props) => <div id="PolarAngleAxis" {...props} />,
  RadialBar: (props) => <div id="RadialBar" {...props} />,
  Tooltip: (props) => <div id="Tooltip" {...props} />,
}));
// ==================================
// unit tests
// ==================================
describe('screen: Home/Consensus', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <Consensus />
      </MockTheme>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
