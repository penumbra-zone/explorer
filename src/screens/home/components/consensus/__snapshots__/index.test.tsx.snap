// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Home/Consensus matches snapshot 1`] = `
<div
  className="makeStyles-root"
  id="box"
>
  <h2
    className="MuiTypography-root makeStyles-label MuiTypography-h2"
  >
    consensus
  </h2>
  <div
    className="makeStyles-info"
  >
    <div>
      <div
        className="MuiTypography-root label MuiTypography-caption"
      >
        height
      </div>
      <div
        className="MuiTypography-root label MuiTypography-caption"
      >
        proposer
      </div>
    </div>
    <div>
      <h4
        className="MuiTypography-root MuiTypography-h4"
      >
        100,500
      </h4>
      <div
        address="Forbole"
        id="AvatarName"
        name="Forbole"
      />
    </div>
  </div>
  <div
    className="makeStyles-content"
  >
    <div
      barSize={10}
      className="makeStyles-chart"
      cx={100}
      cy={100}
      data={
        Array [
          Object {
            "fill": "#FD5D4E",
            "value": 73,
          },
        ]
      }
      endAngle={-270}
      height={200}
      id="RadialBarChart"
      innerRadius={90}
      outerRadius={90}
      startAngle={90}
      width={200}
    >
      <div
        angleAxisId={0}
        domain={
          Array [
            0,
            100,
          ]
        }
        id="PolarAngleAxis"
        tick={false}
        type="number"
      />
      <div
        background={true}
        cornerRadius={100}
        dataKey="value"
        id="RadialBar"
      />
      <div
        id="Tooltip"
      />
      <text
        className="progress-label"
        dominantBaseline="middle"
        textAnchor="middle"
        x={100}
        y={100}
      >
        <tspan
          className="makeStyles-chartPercentLabel"
        >
          73%
        </tspan>
      </text>
      <text
        className="makeStyles-chartExtraLabel"
        x={68}
        y={135}
      >
        <tspan
          className="makeStyles-chartLabel"
        >
          Round 0
        </tspan>
      </text>
      <text
        className="makeStyles-chartExtraLabel"
        x={76}
        y={155}
      >
        <tspan
          className="makeStyles-chartLabel"
        >
          Step 1
        </tspan>
      </text>
    </div>
  </div>
</div>
`;
